### ISSUE TITLE
Something wrong with the Joomla error handler

### ISSUE THREAD

**Issue by philip-sorokin** [joomla/joomla-cms#15215]
I use Nginx + php7.0-fpm. Some errors are not visible to the system - error log is clear.

### Steps to reproduce the issue

Add the following string in the `index.php` file right after the application initialization:

```
// Instantiate the application.
$app = JFactory::getApplication('site');

$nonexistent = null;
echo $nonexistent->nonexistent() + $abcd;
```

### Expected and actual result
You will see the Joomla! error page (see the screenshot - with a strange error **0**), but in the error log you will not find the error. But if you write the following:

```
// Instantiate the application.
$app = JFactory::getApplication('site');
echo $nonexistent;
```
You will see the notice in the log.

### System information (as much as possible)

Debian 8.
Nginx + php7.0-fpm.
Joomla! RC2.
Error reporting: development (the same with `none`).

![err](https://cloud.githubusercontent.com/assets/11632367/24892626/bc6ec11c-1e86-11e7-9f02-2f162370fbcc.jpg)

P.S. I discovered such behaviour when I was debugging a component, but it was impossible to find an error, because the error log was clear, and the Joomla! error page does not point to the broken file and string number -- It just gives a strange **error 0** without any localization.

**Comment by alikon** [comment]
What happens with the system debug plugin enabled  ?

**Comment by philip-sorokin** [comment]
The plugin is not showing at all!

I assume, it is because of the fatal error. If you add the wrong code at the beginning of the file, you can see in the log:

> 2017/04/11 17:51:40 [error] 508#508: *4755 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error: Call to a member function nonexistent() on null in /home/ozon/web/dev.addondev.com/index.php:19

But when it goes after the application instantiation, you have an error 0 and the clear error log.

**Comment by mbabker** [comment]
Joomla doesn't do anything to alter PHP's internal error logging, unless it is affected by us changing display_errors or calling `error_reporting()`, in which case it would still be coupled to PHP's configuration.

**Comment by philip-sorokin** [comment]
@mbabker 
If you write wrong code at the beginning of the index.php file, you will have a fatal error logged. When adding the same wrong code after the application instantiation, you will have an error 0 and the clear log. My configuration is standard PHP7.0-fpm configuration, the same as Nginx.

**Comment by mbabker** [comment]
Depends on the type of error thrown.  PHP 7 Throwables will get caught in Joomla's exception handling layer, and this includes some things in PHP 5 that would trigger notices, warnings, and errors.

**Comment by philip-sorokin** [comment]
@mbabker, I mean the same error. I'll explain again.

Add the wrong code at the beginning of the index.php

```
define('JOOMLA_MINIMUM_PHP', '5.3.10');

$nonexistent = null;
echo $nonexistent->nonexistent() + $abcd;
```

You will have an error logged

> 2017/04/11 18:14:56 [error] 509#509: *4875 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error: Call to a member function nonexistent() on null in /home/ozon/web/dev.addondev.com/index.php:15

![true-500](https://cloud.githubusercontent.com/assets/11632367/24916686/8302de10-1ee3-11e7-9328-d03b60f388da.jpg)

Then add the same code after Joomla instantiate the application:

```
// Instantiate the application.
$app = JFactory::getApplication('site');

$nonexistent = null;
echo $nonexistent->nonexistent() + $abcd;
```
The fatal error becomes **error 0** and does not go into the log. And in the browser console you have the error description:

![wrong-500](https://cloud.githubusercontent.com/assets/11632367/24916760/c3cb5ddc-1ee3-11e7-96c4-c9670efe8193.jpg)

The above shows another issue. Why Joomla! demonstrates the server errors to everyone? It is a security issue, because you cannot disable it, even if you set the error_reporting param as **none**

**Comment by franz-wohlkoenig** [comment]
closed as having PR #15994<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15215">issues.joomla.org/tracker/joomla-cms/15215</a>.</sub>
